src/Symfony/Component/Form/Extension/PasswordHasher/EventListener/PasswordHasherListener.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form\Extension\PasswordHasher\EventListener;
+
+use Symfony\Component\Form\Exception\InvalidConfigurationException;
+use Symfony\Component\Form\Extension\Core\Type\RepeatedType;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\PasswordHasher\Hasher\UserPasswordHasherInterface;
+use Symfony\Component\PropertyAccess\PropertyAccess;
+use Symfony\Component\PropertyAccess\PropertyAccessorInterface;
+use Symfony\Component\Security\Core\User\PasswordAuthenticatedUserInterface;
+
+/**
+ * @author Sébastien Alfaiate <s.alfaiate@webarea.fr>
+ */
+class PasswordHasherListener
+{
+    private array $passwords = [];
+
+    public function __construct(
```

Comment:
```suggestion
    public function __construct(
        private PasswordHasherFactoryInterface $hasherFactory,
        // ...
    }
    
    public function registerPassword(FormEvent $event)
    {
        // ...
        
        if (!$parentForm || !$user = $parentForm->getData() || !$user instanceof PasswordAuthenticatedUserInterface || !$user instanceof PasswordHasherAwareInterface) {
        // ...
    }
    
    public function hashPasswords(FormEvent $event)
    {
        // ...
                    $password['user'],
                    $password['property_path'],
                    $this->hashPassword($password['user'], $password['password'])
        // ...
    }
    
    private function hashPassword(PasswordAuthenticatedUserInterface|PasswordHasherAwareInterface $user, $plainPassword): string
    {
        $salt = null;
        if ($user instanceof LegacyPasswordAuthenticatedUserInterface) {
            $salt = $user->getSalt();
        }

        $hasher = $this->hasherFactory->getPasswordHasher($user);

        return $hasher->hash($plainPassword, $salt);
    }
```

Reply:
This will be partially broken.

If your DTO class only implements `PasswordHasherAwareInterface` and `getPasswordHasherName()` returns a legacy hashing method requiring the `$salt` parameter, it will not work (how to get the salt?).
Unless we add a new `hash_salt` option as I suggested in my previous comment.

The `getSalt()` method is only part of the `LegacyPasswordAuthenticatedUserInterface`.